xonsh/aliases.py

```diff
@@ -109,6 +114,22 @@ def source_bash(args, stdin=None):
     return source_foreign(args, stdin=stdin)
 
 
+def source_cmd(args, stdin=None):
+    """Simple cmd.exe-specific wrapper around source-foreign."""
+    args = list(args)
+    fpath = locate_binary(args[0])
+    args[0] = fpath if fpath else args[0]
+    args.insert(0, 'cmd')
+    args.append('--interactive=0')
+    args.append('--sourcer=call')
+    args.append('--funcscmd=echo.')
+    args.append('--aliascmd=echo.')
+    args.append('--envcmd=set')
+    args.append('--safe=0')
+
+    return source_foreign(args, stdin=stdin)
```

Comment:
Ahh. I can do this. I just have a some code in the function that locates the full path of the file we are sourcing. I will try to see if I can do without it.